pupil_src/shared_modules/video_export/plugins/world_video_exporter.py

```diff
@@ -0,0 +1,292 @@
+        )
+
+        # setup of writer
+        writer = AV_Writer(
+            out_file_path, fps=cap.frame_rate, audio_loc=audio_path, use_timestamps=True
+        )
+
+        cap.seek_to_frame(start_frame)
+
+        start_time = time()
+
+        g_pool.plugin_by_name = plugin_by_name
+        g_pool.capture = cap
+        g_pool.rec_dir = rec_dir
+        g_pool.user_dir = user_dir
+        g_pool.meta_info = meta_info
+        g_pool.timestamps = timestamps
+        g_pool.delayed_notifications = {}
+        g_pool.notifications = []
+
+        for initializers in pre_computed_eye_data.values():
+            initializers["data"] = [
+                fm.Serialized_Dict(msgpack_bytes=serialized)
+                for serialized in initializers["data"]
+            ]
+
+        g_pool.pupil_positions = pm.Bisector(**pre_computed_eye_data["pupil"])
+        g_pool.gaze_positions = pm.Bisector(**pre_computed_eye_data["gaze"])
+        g_pool.fixations = pm.Affiliator(**pre_computed_eye_data["fixations"])
+
+        # add plugins
+        g_pool.plugins = Plugin_List(g_pool, plugin_initializers)
+
+        while frames_to_export > current_frame:
+            try:
+                frame = cap.get_frame()
+            except EndofVideoError:
+                break
+
+            events = {"frame": frame}
+            # new positions and events
+            frame_window = pm.enclosing_window(g_pool.timestamps, frame.index)
+            events["gaze"] = g_pool.gaze_positions.by_ts_window(frame_window)
+            events["pupil"] = g_pool.pupil_positions.by_ts_window(frame_window)
+
+            # publish delayed notifications when their time has come.
+            for n in list(g_pool.delayed_notifications.values()):
+                if n["_notify_time_"] < time():
+                    del n["_notify_time_"]
+                    del g_pool.delayed_notifications[n["subject"]]
+                    g_pool.notifications.append(n)
+
+            # notify each plugin if there are new notifications:
+            while g_pool.notifications:
+                n = g_pool.notifications.pop(0)
+                for p in g_pool.plugins:
+                    p.on_notify(n)
+
+            # allow each Plugin to do its work.
+            for p in g_pool.plugins:
+                p.recent_events(events)
+
+            writer.write_video_frame(frame)
+            current_frame += 1
+            yield "Exporting with pid {}".format(PID), current_frame
+
+        writer.close()
+
+        duration = time() - start_time
+        effective_fps = float(current_frame) / duration
+
+        result = "Export done: Exported {} frames to {}. This took {} seconds. Exporter ran at {} frames per second."
+        print(result.format(current_frame, out_file_path, duration, effective_fps))
+        yield "Export done. This took {:.0f} seconds.".format(duration), current_frame
+
+    except GeneratorExit:
+        print("Video export with pid {} was canceled.".format(os.getpid()))
```

Comment:
Could you please replace all print statements with `logger.debug/warning/error()` appropriately? We did not use logger before because of #1305

Reply:
Roger